### PR TITLE
Adding support for additional authentication options in client

### PR DESCRIPTION
**Pull request by wolfejw86**
Addresses #55

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,6 +106,44 @@ If you need to use basic authentication to connect with the Elasticsearch cluste
...
+  auth: {
+    username: 'user',
+    password: 'pwd'
+  },
+  'es-version': 7,
+  'flush-bytes': 1000
+})
+```
+
+And if you're using elastic cloud, you can pass an `apiKey` instead:
```
Comment (delvedor) [inline README.md:127]: This is not entirely correct, you can use api key also without elastic cloud and you can use basic auto with elastic cloud as well :)

Comment (wolfejw86) [reply README.md:139]: Ah, shows my lack of comprehensive knowledge of elastic! Thank you for pointing this out.

**Comment by wolfejw86** [comment]
@delvedor I added support for the cli as well. TBH I haven't done a lot of CLI work. I attempted to add the parameters in a way that I thought made sense, but am definitely open to feedback on that portion of the implementation.  Thanks for the quick review!

**Comment by wolfejw86** [comment]
@mcollina yes I can do that. I hadn't used proxyquire before but I saw it in there and feel like I can just follow the pattern for how ya'll are already using it 👍

**Comment by wolfejw86** [comment]
@mcollina do the tests I've added target what you're after? (I'm sure it's been a busy few days with the OpenJS conference - I enjoyed your talk btw. )
If not, any feedback on how to improve their efficacy would be much appreciated.  Thanks!

**Comment by mcollina** [comment]
Good for me... @delvedor wdyt?

**Comment by wolfejw86** [comment]
@delvedor Please let me know if there's anything else you think this needs :) 

Thanks!

**Comment by wolfejw86** [comment]
Is there anything we're waiting on for this to be merged?

**Comment by mcollina** [comment]
A review by @delvedor. I pinged him again, let's see otherwise we'll land in a few days.
Sorry for the update.

**Pull request merged**